src/app/task/shared/services/task.service.ts

```diff
@@ -1,36 +1,33 @@
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { Task } from "../models/task.model";
+import { Observable } from "rxjs";
 
 @Injectable({
   providedIn: 'root'
 })
 export class TaskService {
-  taskList : Task[] = new Array();
+  BASE_URL = 'http://localhost:3000/api/tasks/';
 
-  constructor(private http: HttpClient) {
-    this.http.get('assets/tasks.json').subscribe((data : any) => {
-      this.taskList = data;
-    });
-  }
+  constructor(private http: HttpClient) {}
 
-  getAllTasks() {
-    return this.taskList;
+  getAllTasks(): Observable<any> {
```

Comment:
change type from any to respective type that's being received (in this case its a list of tasks)